kuka_rox_hardware_interface/src/robot_manager_node.cpp

```diff
@@ -229,50 +239,49 @@ RobotManagerNode::on_activate(const rclcpp_lifecycle::State &)
     return FAILURE;
   }
 
-  // Activate joint state broadcaster
-  auto controller_request =
-    std::make_shared<SwitchController::Request>();
-  controller_request->strictness = SwitchController::Request::STRICT;
-  controller_request->activate_controllers = std::vector<std::string>{"joint_state_broadcaster"};
-  auto controller_response =
-    kroshu_ros2_core::sendRequest<SwitchController::Response>(
-    change_controller_state_client_, controller_request, 0, 2000
-    );
-  if (!controller_response || !controller_response->ok) {
-    RCLCPP_ERROR(get_logger(), "Could not start joint state broadcaster");
-    // TODO(Svastits): this can be removed if rollback is implemented properly
-    this->on_deactivate(get_current_state());
-    return FAILURE;
-  }
+    // Select controllers
+    auto control_mode = this->get_parameter("control_mode").as_int();
+    auto new_controllers = controller_handler_.GetControllersForSwitch(
+      kroshu_ros2_core::ControllerHandler::ControlMode(
+        control_mode));
+
+    // Activate RT controller(s)
+    auto controller_request =
+      std::make_shared<SwitchController::Request>();
+    controller_request->strictness = SwitchController::Request::STRICT;
+    controller_request->activate_controllers = new_controllers.first;
+    if (!new_controllers.second.empty()) {
+      // This shuld never happen
+      controller_request->deactivate_controllers = new_controllers.second;
+      RCLCPP_WARN(get_logger(), "Deactivating controllers while activating");
```

Comment:
This log message still does not say anything to a user
"Controller handler state is improper, as active controller list is not empty before activation" would be better
This should indeed never happen, but as rollback is not yet perfectly implemented, I would also make the severity ERROR